src/translator/service.h

```diff
@@ -40,16 +62,18 @@ class Service {
   /// @param options Marian options object
   /// @param model_memory byte array (aligned to 64!!!) that contains the bytes
   /// of a model.bin. Optional, defaults to nullptr when not used
-  explicit Service(Ptr<Options> options, const void *model_memory = nullptr);
+  explicit Service(Ptr<Options> options, const MemoryGift* modelMemory, const MemoryGift* shortlistMemory);
+
+  explicit Service(Ptr<Options> options) : Service(options, nullptr, nullptr){}
 
-  /// Construct Service from a string configuration.
+/// Construct Service from a string configuration.
   /// @param [in] config string parsable as YAML expected to adhere with marian
   /// config
   /// @param [in] model_memory byte array (aligned to 64!!!) that contains the
   /// bytes of a model.bin. Optional, defaults to nullptr when not used
   explicit Service(const std::string &config,
-                   const void *model_memory = nullptr)
-      : Service(parseOptions(config), model_memory) {}
+                   const void* modelMemory = nullptr, const void* shortlistMemory = nullptr)
+      : Service(parseOptions(config), hackModel(modelMemory), hackShortLis(shortlistMemory)) {}
```

Comment:
I want to be picky about `hackShotLis`, but won't insist since this function is meant to go away anyway.